Scaffold `lib/jormungandr` folder

Anviking opened the pull request:
# Issue Number

#218 


# Overview

<!-- Detail in a few bullet points the work accomplished in this PR -->

- [x] I added `lib/jormungandr` based upon `lib/http-bridge`
- [x] Most notably there is `{Binary,Compatibility,Transaction}.Jormungandr.hs`, and `Cardano.Wallet.Binary.JormungandrSpec`

# Comments

- Ah 🤦‍♂ I think the new format is not even CBOR. I have now started from a cleaner slate than initially.

<!-- Additional comments or screenshots to attach if any -->

<!-- 
Don't forget to:

 ✓ Self-review your changes to make sure nothing unexpected slipped through
 ✓ Assign yourself to the PR
 ✓ Assign one or several reviewer(s)
 ✓ Once created, link this PR to its corresponding ticket
 ✓ Acknowledge any changes required to the Wiki
-->

Anviking committed:
Add stub cardano-wallet-jormungandr

Anviking commented on lib/jormungandr/src/Cardano/Environment.hs:
```diff
@@ -0,0 +1,166 @@
+-- rely on a `.env` file to bundle configuration settings together for a given
+-- target environment.
+
+module Cardano.Environment
```
I wasn't sure moving `lib/{jormungandr, http-bridge}/Cardano/Environment.hs` to `cardano-wallet-core` was desired, so I kept the duplication for now.

KtorZ replied:
Well, network discrimination is different on `Jörmungandr`, and probably will be too with Shelley Haskell. There are actually only too network (testnet or mainnet) and it's encoded on a single byte (Cf the new address format). Perhaps it'd make sense to rename both modules to `Cardano.Environment.{Jormungandr,HttpBridge}`.

KtorZ replied:
Yet, we could factor out in `core` the "Internal" bits with the `ErrMissingOrInvalidEnvVar` and `unsafeLookupEnv` since this is fairly network-agnostic.

KtorZ reviewed (approved):
Thanks for this. I think we can factor our some of the logic in `Cardano.Environment` (to `core`), and rename both modules in the bridge and jormungandr to implement the specific bits.

Anviking committed:
Rename to Cardano.Environment.{HttpBridge, Jormungandr}

Anviking committed:
Factor out unsafeLookupEnv

- Added System.Environment.Lookup and System.Environment.LookupSpec to
core

- Removed common parts from
    - Cardano.Environment.HttpBridge
    - Cardano.Environment.HttpBridgeSpec
    - Cardano.Environment.Jormungandr
    - Cardano.Environment.JormungandrSpec (added)

akegalj commented on lib/jormungandr/src/Cardano/Environment/Jormungandr.hs:
```diff
@@ -0,0 +1,70 @@
+-- configuration data at runtime using the available environment.
+--
+-- This gives us a flexible and portable approach to software configuration, and
+-- remove some pain from the development perspective. Prior to starting, the
```
s/remove/removes

plural

akegalj reviewed (commented):
lgtm
